pkg/ingester/stream.go

```diff
@@ -227,6 +231,18 @@ func (s *stream) Push(
 			chunk = s.cutChunk(ctx)
 		}
 
+		// Check if this this should be rate limited.
+		now := time.Now()
+		if !s.limiter.AllowN(now, s.tenant, len(entries[i].Line)) {
+			validation.DiscardedSamples.WithLabelValues(validation.StreamRateLimit, s.tenant).Add(float64(len(entries) - i))
```

Comment:
Please use accumulator counters for the discarded samples, similar to how `out_of_order`s are handled on
line 206-212. This is the hot path and it's nice to call this once per batch rather than once per sample.